src/bin/customer/establish.rs

```diff
@@ -392,6 +392,10 @@ async fn get_parameters(
         .await
         .context("Failed to receive merchant's range proof parameters")?;
 
+    if range_constraint_parameters.validate().is_err() {
+        return Err(establish::Error::InvalidParameters.into());
+    }
+
```

Comment:
Nit, but can you remove the comment on L415-416? Perhaps also L414 and 417, since there is implicit validation in the deserialization of those types.